src/main/java/com/systemmeltdown/meltdownlog/LogSession.java

```diff
@@ -0,0 +1,90 @@
+package com.systemmeltdown.meltdownlog;
+
+import com.systemmeltdown.meltdownlog.lib.NanoTimeReference;
+import com.systemmeltdown.meltdownlog.outputs.LogOutput;
+import com.systemmeltdown.meltdownlog.topics.LogTopic;
+import com.systemmeltdown.meltdownlog.topics.LogTopicRegistry;
+
+/**
+ * Represents a time-based session of logging. Each session is pre-configured
+ * and logs to its outputs after it's started until it is stoppped.
+ */
+public abstract class LogSession implements NanoTimeReference {
+	protected final LogTopicRegistry m_topicRegistry;
+	protected long m_startNanos = Long.MIN_VALUE;
+	protected long m_stopNanos = Long.MIN_VALUE;
+
+	public LogSession(final LogTopicRegistry topicRegistry) {
```

Comment:
Why doesn't this use the singleton instance of LogTopicRegistry?

Reply:
It will, I just hadn't added it yet because I was only doing tests with this. I'll add it now.